manifests/record/a.pp

```diff
@@ -19,15 +19,10 @@
     data => $data
   }
 
-  if $ptr {
+  if $ptr == 'all' {
+    dns::record::ptr::by_ip { $data: host => $host, zone => $zone }
```

Comment:
It might have been more readable for me to break it into multiple lines:

    dns::record::ptr::by_ip { $data:
      host => $host ,
      zone => $zone ,
    }

`$data` is the array of ip addresses that was passed into the `dns::record::a` type.  I'm passing it as the array of titles for the `dns::record::ptr::by_ip` type to let the defined type invocation do the iteration for me.